packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java

```diff
@@ -124,18 +104,52 @@ public void run() {
   @Override
   public void clearFocus() {
     super.clearFocus();
+    resetInputConnection();
+  }
+
+  /**
+   * Ensure that input creation happens back on {@link #containerView}'s thread.
+   *
+   * <p>The logic in {@link #checkInputConnectionProxy} forces input creation to happen on Webview's
+   * thread for all connections. We undo it here so users will be able to go back to typing in
+   * Flutter UIs as expected.
+   */
+  private void resetInputConnection() {
     if (proxyAdapterView == null) {
       // No need to reset the InputConnection to the default thread if we've never changed it.
       return;
     }
-    containerView.requestFocus();
+    setInputConnectionThread(/*targetView=*/ containerView);
+  }
+
+  /**
+   * This is the crucial trick that gets the InputConnection creation to happen on the correct
+   * thread.
+   * https://cs.chromium.org/chromium/src/content/public/android/java/src/org/chromium/content/browser/input/ThreadedInputConnectionFactory.java?l=169&rcl=f0698ee3e4483fad5b0c34159276f71cfaf81f3a
+   *
+   * <p>{@code targetView} should have a {@link View#getHandler} method with the thread that future
+   * InputConnections should be created on.
+   */
+  private void setInputConnectionThread(final View targetView) {
```

Comment:
nit: Maybe call it setInputConnectionTarget? (as the parameter is the target which also specified the thread, and not the thread itself.)

Reply:
Done.